### DISCUSSION TITLE
Integration Suggestion + Random Ideas

### DISCUSSION THREAD

**Discussion by Turkey9002** [SirSquidly/Oceanic-Expanse#3] [Ideas]
First of all this is a super great mod so far! My favorite additions are clams, crabs, and coconuts. :) I'm hoping there's integration with subaquatic in the future, I'm aware that this mod was worked on since 2018 before that mod even came out and I did noticed that pretty much everything is configurable which is excellent! 
It's not possible to play with the new generated features on subaquatic's biomes though (i.e. sea stars, dulse, and tube sponge on frozen oceans) and both mods handle their fish buckets differently. I think a way to detect it and then automatically disable conflicting features while integrating into that mod would be great, but another more harder way would be to make the mod depend on subaquatic instead (I don't know how frustrating that would be though x.x).

I also have some random ideas, I'm pretty picky with what I suggest too so hopefully it's not a quantity over quality thing.

Fish:
--
- **Leopard Shark:** Sharks are heavily requested I know, I think a great way to implement them though is by adding one of the most peaceful sharks. Leopard Sharks are a neutral mob (they'll attack you if you attack them) and they'd be found mainly lurking at the bottom of deep warm oceans and deep lukewarm oceans (with subaquatic). They'll sometimes hunt down random fish when hungry.
- **Suckermouth Catfish:** Bottom feeders commonly found in rivers, jungles, swamps, and all oceans except frozen (rather infamous for being invasive in real life but they're one of my favorites).

Hostile Mobs:
--
- **Monster of the Ocean Depths (Barnacle):** I'm still upset about how the first mob vote turned out since this is the mob I wanted in the most outside of the Hovering Inferno. I recommend checking out Outvoted (it's GNU licensed) for code, the model, and some cool sounds if this is interesting enough. I think for mechanics it should spawn rather rarely (as a sense of surprise), it definitely should not pull you if you're riding a boat and it should stay hidden in dark areas. It could drop it's tongue which can be used to make a grapple or a whip of some sort.
- **Sunken (Coral) Skeleton:** The very same ones featured in Minecraft Dungeons, it's a simple mob acting like an underwater variant of the skeleton. Instead of crossbows, they should use harpoons instead in a similar matter to tridents. Additionally they can work well as a renewable source of coral (uncommon drop).

Misc:
--
- **Crab Texture Variants:** They'd be perfect with more texture variants like the Quark crab in modern versions.
- **Craftable Sponge:** 3x3 of tube sponges should make a sponge block.
- **Guardian Spikes:** A redstone signal should retract the spikes, doesn't exactly have to be like what Abnormals did but it would be very good for redstone mechanisms. Would also love to see an elder version, you could make it different by having it always damage you when touched unlike the regular version (unless retracted).

**Comment by SirSquidly** [comment]
Damn, I'll be honest, a ton of these are brilliant suggestions! So, first on integration:

To be honest, I want to keep this mod from requiring any other mods, which is why Subaquatic Integration will be a bit tough to figure out, if I want to do it at all. I have considered it, though I might want to wait and see the mod reach further in development. Consider it extremely possible.

On suggestions:

Leopard Shark: Not a bad mob to add for ambience, although I have many other mobs I have planned that take priority.

Catfish: So, catfish were actually planned very early on in this mod, and I even have a model sitting around made for them. I cut them due to time and the fact they are a bit outside the scope of the mod (Ocean content), though they might be brought back in the future.

Barnacle: Another mob I fully planned, and is extremely likely to be added later. I have a lot of ideas for this mob.

Sunken Skeleton: This and the Tropical Slime are both likely to be added later, when I can think of some good mechanics for them. I'm brainstorming it.

Crab Textures: Expect this to be a feature later, as I wanted to make crabs come in some skin variants, including unique ones per biome. Like rabbits.

Craftable Sponge: This was an oversight on my part. I fully intended Tube Sponge to be crafted into Sponge, I, uh... forgot? Yeah, this is a 100% guaranteed addition.

Guardian Spike: I feel like retracting the spike isn't too necessary. 
As for Elder Guardian Spikes, I feel like they miss the purpose of the Spikes, being primarily for crafting the Trident. Each elder drops 1 spike guaranteed, so 1 monument = 1 trident, while normal Guardians can also be farmed for spikes. To add a sperate spike for the Elder would be odd for this purpose, as would it make a different trident? If it does, then it would have an odd inverse where it's easier to get than a normal trident, but non-renewable. If not, then you are just spending a non-renewable block for a renewable item. It complicates things.

**Reply by Turkey9002** [reply]
Most excited to see what you'll do with the dungeons mobs and especially the Barnacle! Hopefully someday the Hovering Inferno and Great Hunger will be added to 1.12 to join in as well, those mobs from the 2017 vote deserved way better. I'm a bit confused with the catfish though, is it due to it not spawning in oceans? You could try changing it to a saltwater species of catfish instead if that's what you mean

I gotcha on how you feel about the spikes, it indeed wouldn't make sense to add an elder version if the regular spike is already used to make a trident, I did had redstone in mind when thinking about them too so that's why I suggested the retraction thing.
